src/ThreeJsPanel.ts

```diff
@@ -27,6 +28,14 @@ const DEFAULT_PERSPECTIVE_CAMERA_FAR = 20.0;
 
 const DEFAULT_ORTHO_SCALE = 0.5;
 
+export type CameraTransform = {
```

Comment:
"transform" seems like an odd name for this type - it makes me think of a "transformation matrix" rather than the decomposed settings captured by this object - but I'm having a hard time coming up with a better one... "Position" isn't general enough to capture everything this type contains, "settings" is too general... 🤷

Reply:
I agree that camera transform isn't quite right.  There are a few weird things here.
1. Just be aware: having rotation and target exist independently means they can become inconsistent, which is bad, but it also is true that "target" is overspecifying rotation.  The location of target is for the camera CONTROLS, not just the camera view.    Basically if you provide position and target, then rotation is fully determined. But ultimately, I guess it's ok to provide rotation and target separately.. I might mull this over a bit longer.
2. For perspective cameras, you need to capture the field of view angle too.  Put a comment or something to say whether it's half the fov or the full fov, and whether it describes horizontal or vertical fov. (Or put it in the name like fullVerticalFovDegrees - overkill....)
3. Is orthoScales supposed to be representing the 3 separate ortho cameras?  Or will you record 4 different CameraTransform objects, one for each of the 4 viewing cameras?  This feels weirdly overloaded.

I'd call this data structure, with the fov angle added, something more like CameraData, CameraInfo, or Camera.

Reply:
Part of my comment above is wondering whether you mean to capture only the active viewing camera, or cameras for all other view modes too.

Reply:
@3: `orthoScales` captures the current camera scales but the rest of the CameraTransform only captures details for the currently active camera. Do you think it's important that we sync all four cameras when we share links?

Reply:
Well I think we should or we shouldn't.  No point in capturing orthoScales unless the current camera is ortho, and then you only need one of them right?  Most important is to capture the "current" view. So I don't think it's super critical to save all 4 cameras.   I suggest making the data structure for all the data needed to record one single camera, and then later on if we want to save them for multiple views ,we save whole entire cameras.

Reply:
Also great point on target + rotation being redundant! I originally only stored position + rotation and it led to some bugged camera behavior until I also added in the target value-- I didn't go back to check if I could remove the rotation after.